### PR TITLE
refactor(components): [el-message][el-notification] improve type

### LINKED ISSUE #3640: [Bug Report] import { ElMessage } from "element-plus";   TS type prompt error
### Element Plus version
1.1.0-beta.16

### OS/Browsers version
win10

### Vue version
3.2.19

### Reproduction Link
https://github.com/element-plus/element-plus

### Steps to reproduce
安装elementui 1.1.0-beta.16   在ts文件中使用使用 
import { ElMessage } from "element-plus";

使用ElMessage 后
提示

此表达式不可调用。
  "SFCWithInstall<Required<MessagePartial>>" 类型的部分要素不可调用。
    类型 "Required<MessagePartial> & { install: PluginInstallFunction; }" 没有调用签名。ts(2349)

### What is Expected?
无错误

### What is actually happening?
ts类型错误

### PR DESCRIPTION
**Pull request by sxzz**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/message/src/message.ts [hunk 1]
```diff
@@ -72,16 +72,15 @@ export interface MessageHandle {
...
-  warning?: (options?: MessageParamsTyped) => MessageHandle
-  info?: (options?: MessageParamsTyped) => MessageHandle
-  error?: (options?: MessageParamsTyped) => MessageHandle
 }
-export type Message = Required<MessagePartial>
+export interface Message extends MessageFn {
+  success: (options?: MessageParamsTyped) => MessageHandle
+  warning: (options?: MessageParamsTyped) => MessageHandle
+  info: (options?: MessageParamsTyped) => MessageHandle
+  error: (options?: MessageParamsTyped) => MessageHandle
```
Comment (JeremyWuuuuu) [inline packages/components/message/src/message.ts:82]: **Suggestion:**
```ts
type TypedMessage = (options?: MessageParamsTyped) => MessageHandle

export interface Message extends MessageFn {
  success: TypedMessage
  // ...rest
```

File: packages/components/notification/src/notification.ts [hunk 2]
```diff
@@ -89,16 +89,15 @@ export type NotificationParamsTyped =
...
-  error?: (options: NotificationParamsTyped) => NotificationHandle
-  info?: (options: NotificationParamsTyped) => NotificationHandle
+export type NotifyFn = ((
+  options?: NotificationParams
+) => NotificationHandle) & { closeAll: () => void }
+export interface Notify extends NotifyFn {
+  success: (options?: NotificationParamsTyped) => NotificationHandle
+  warning: (options?: NotificationParamsTyped) => NotificationHandle
+  error: (options?: NotificationParamsTyped) => NotificationHandle
+  info: (options?: NotificationParamsTyped) => NotificationHandle
```
Comment (JeremyWuuuuu) [inline packages/components/notification/src/notification.ts:99]: **Suggestion:**
```ts
type TypedNotification = (options?: NotificationParamsTyped) => NotificationHandle

export interface Notify extends NotifyFn {
  success: TypedMessage
  // ...rest
}
```

File: packages/components/notification/src/notify.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ const notifications: Record<
 const GAP_SIZE = 16
 let seed = 1
 
-const notify: NotifyPartial = function (options = {}) {
+const notify: NotifyFn & Partial<Notify> = function (options = {}) {
```
Comment (JeremyWuuuuu) [inline packages/components/notification/src/notify.ts:32]: **Question: ** Why `Partial`?

**Review by jw-foss — APPROVED** [review]
Left a few non-blocking comments for you to consider, approving as-is, feel free to merge this one.

File: packages/components/notification/src/notify.ts (hunk 3 above)
Comment (sxzz) [reply packages/components/notification/src/notify.ts:32]: TypeScript can not detect typed methods is defined, so make it optional.

**Comment by sxzz** [comment]
Done.

**Pull request merged**